src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -0,0 +1,20 @@
+package lotto.domain
+
+class LottoNumber(val number: Int) {
+
+    init {
+        require(number in MIN_NUMBER..MAX_NUMBER) {
+            "로또 번호는 1 이상 45 이하여야 합니다."
+        }
+    }
+
+    companion object {
+        const val MIN_NUMBER = 1
+        const val MAX_NUMBER = 45
+    }
+
+}
+
+object LottoNumbers {
+    val ALL_LOTTO_NUMBERS = (LottoNumber.MIN_NUMBER..LottoNumber.MAX_NUMBER).map { LottoNumber(it) }
+}
```

Comment:
LottoNumber의 생성을 로또 풀에서만 가능하도록 생성자를 private로 만들고
풀을 외부 공개보단 팩토리에서 풀을 사용하는게 좋을듯합니다.

```suggestion
class LottoNumber private (val number: Int) {

    init {
        require(number in MIN_NUMBER..MAX_NUMBER) {
            "로또 번호는 1 이상 45 이하여야 합니다."
        }
    }

    companion object {
        const val MIN_NUMBER = 1
        const val MAX_NUMBER = 45
        
        private val ALL_LOTTO_NUMBERS = (LottoNumber.MIN_NUMBER..LottoNumber.MAX_NUMBER).map { LottoNumber(it) }
        
        fun create(num:Int){
          return ALL_LOTTO_NUMBERS[num]
        }
    }

}


```